python/mlc_chat/compiler_pass/prune_relax_func.py

```diff
@@ -0,0 +1,24 @@
+"""A pass that removes undesired Relax function from IRModule based on target."""
+import tvm
+from tvm import IRModule
+
+from mlc_chat.support.auto_target import detect_cuda_arch_list
+
+
+@tvm.transform.module_pass(opt_level=0, name="PruneRelaxFunc")
+class PruneRelaxFunc:  # pylint: disable=too-few-public-methods
+    """Removes undesired Relax function from IRModule based on target."""
+
+    def transform_module(self, mod: IRModule, _ctx: tvm.transform.PassContext) -> IRModule:
+        """Entrypoint"""
+        target = tvm.target.Target.current(allow_none=False)
+        func_dict = {}
```

Comment:
it might be the case where the original `IRModule` has extra fields such as `attrs`, which this pass happens to discard - shall we also add them back?

Reply:
Thanks for catching! Will add. It would be simpler if IRModule supports removing function from Python side. Unfortunately we don't have that right now.

Reply:
Some history: we were trying to push immutability of IRModule, but it didn't work out :((